src/System.Text.Encoding.Extensions/tests/fallback.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.Collections;
+using System.Text;
+
+namespace EncodingTests
+{
+    public static class Fallback
+    {
+        static readonly string s_asciiInputStringWithFallback = "\u00abX\u00bb"; // DOUBLE ANGLE QUOTATION MARK (U+00AB), 'X' (U+0058), and RIGHT POINTING DOUBLE ANGLE QUOTATION MARK (U+00BB).
+        static readonly string s_asciiInputStringWinNoFallback = "abc";
+
+        [Fact]
+        public static void TestEncoderReplacementFallback()
+        {
+            Encoding asciiEncoding = Encoding.GetEncoding("us-ascii", new EncoderReplacementFallback("(unknown)"), new DecoderReplacementFallback(""));
+            byte[] encodedBytes = new byte[asciiEncoding.GetByteCount(s_asciiInputStringWithFallback)];
+
+            int numberOfEncodedBytes = asciiEncoding.GetBytes(s_asciiInputStringWithFallback, 0, s_asciiInputStringWithFallback.Length, encodedBytes, 0);
+
+            AssertEqualList(
```

Comment:
I thought Assert.Equal is not testing the collection contents. I searched the code and I saw some other places doing the same as I did. I'll try your idea.